DTrace works terribly with MariaDB. Without this patch DTrace support even can't be compiled.
https://www.illumos.org/issues/4248 
diff -ru mariadb-5.5.32.orig/cmake/plugin.cmake mariadb-5.5.32/cmake/plugin.cmake
--- mariadb-5.5.32.orig/cmake/plugin.cmake	2013-07-17 18:51:28.000000000 +0400
+++ mariadb-5.5.32/cmake/plugin.cmake	2013-11-11 13:37:44.978143666 +0400
@@ -134,7 +134,7 @@
     ENDIF()
 
     ADD_LIBRARY(${target} STATIC ${SOURCES})
-    DTRACE_INSTRUMENT(${target})
+#    DTRACE_INSTRUMENT(${target})
     ADD_DEPENDENCIES(${target} GenError ${ARG_DEPENDENCIES})
     RESTRICT_SYMBOL_EXPORTS(${target})
     IF(WITH_EMBEDDED_SERVER)
@@ -143,7 +143,7 @@
       IF(ARG_RECOMPILE_FOR_EMBEDDED OR NOT _SKIP_PIC)
         # Recompile some plugins for embedded
         ADD_CONVENIENCE_LIBRARY(${target}_embedded ${SOURCES})
-        DTRACE_INSTRUMENT(${target}_embedded)   
+#        DTRACE_INSTRUMENT(${target}_embedded)   
         IF(ARG_RECOMPILE_FOR_EMBEDDED)
           SET_TARGET_PROPERTIES(${target}_embedded 
             PROPERTIES COMPILE_DEFINITIONS "EMBEDDED_LIBRARY")
@@ -182,7 +182,7 @@
   
     ADD_VERSION_INFO(${target} MODULE SOURCES)
     ADD_LIBRARY(${target} MODULE ${SOURCES}) 
-    DTRACE_INSTRUMENT(${target})
+#    DTRACE_INSTRUMENT(${target})
     SET_TARGET_PROPERTIES (${target} PROPERTIES PREFIX ""
       COMPILE_DEFINITIONS "MYSQL_DYNAMIC_PLUGIN")
     TARGET_LINK_LIBRARIES (${target} mysqlservices)
--- mariadb-5.5.32.orig/sql/CMakeLists.txt	2013-07-17 18:51:28.000000000 +0400
+++ mariadb-5.5.32/sql/CMakeLists.txt	2013-11-11 16:23:16.414591695 +0400
@@ -97,10 +97,22 @@
 
 MYSQL_ADD_PLUGIN(partition ha_partition.cc STORAGE_ENGINE DEFAULT STATIC_ONLY
 RECOMPILE_FOR_EMBEDDED)
+DTRACE_INSTRUMENT(partition)
+IF(WITH_EMBEDDED_SERVER)
+  IF(ARG_RECOMPILE_FOR_EMBEDDED OR NOT _SKIP_PIC)
+    DTRACE_INSTRUMENT(partition_embedded)
+  ENDIF()
+ENDIF()
 
 ADD_LIBRARY(sql STATIC ${SQL_SOURCE})
 ADD_DEPENDENCIES(sql GenServerSource)
 DTRACE_INSTRUMENT(sql)
+IF(WITH_EMBEDDED_SERVER)
+  IF(ARG_RECOMPILE_FOR_EMBEDDED OR NOT _SKIP_PIC)
+    DTRACE_INSTRUMENT(sql_embedded)
+  ENDIF()
+ENDIF()
+
 TARGET_LINK_LIBRARIES(sql ${MYSQLD_STATIC_PLUGIN_LIBS} 
   mysys dbug strings vio regex
   ${LIBWRAP} ${LIBCRYPT} ${LIBDL}
--- mariadb-5.5.32.orig/storage/csv/CMakeLists.txt	2013-07-17 18:51:28.000000000 +0400
+++ mariadb-5.5.32/storage/csv/CMakeLists.txt	2013-11-11 16:21:54.747257748 +0400
@@ -15,3 +15,9 @@
 
 SET(CSV_SOURCES  ha_tina.cc ha_tina.h transparent_file.cc transparent_file.h)
 MYSQL_ADD_PLUGIN(csv ${CSV_SOURCES} STORAGE_ENGINE MANDATORY)
+DTRACE_INSTRUMENT(csv)
+IF(WITH_EMBEDDED_SERVER)
+  IF(ARG_RECOMPILE_FOR_EMBEDDED OR NOT _SKIP_PIC)
+    DTRACE_INSTRUMENT(csv_embedded)
+  ENDIF()
+ENDIF()
